python_modules/dagster/dagster/_core/workspace/load_target.py

```diff
@@ -41,24 +42,95 @@ def create_origins(self) -> Sequence[CodeLocationOrigin]:
         return location_origins_from_yaml_paths(self.paths)
 
 
-def get_origins_from_toml(path: str) -> Sequence[ManagedGrpcPythonEnvCodeLocationOrigin]:
+def validate_dagster_block_for_module_name_or_modules(dagster_block):
+    module_name_present = "module_name" in dagster_block and isinstance(
+        dagster_block.get("module_name"), str
+    )
+    modules_present = "modules" in dagster_block and isinstance(
+        dagster_block.get("modules"), list
+    )
+
+    if module_name_present and modules_present:
+        # Here we have the check only for list; to be a bit more forgiving in comparison to 'is_valid_modules_list' in case it's an empty list next to 'module_name' existance
+        if len(dagster_block["modules"]) > 0:
+            raise ValueError(
+                "Only one of 'module_name' or 'modules' should be specified, not both."
+            )
+
+    if modules_present and len(dagster_block.get("modules")) == 0:
+        raise ValueError("'modules' list should not be empty if specified.")
+
+    return True
+
+
+def is_valid_modules_list(modules) -> bool:
```

Comment:
type hint for modules would be nice

Reply:
Got it, and added List[Dict[str, str]] as I think stricter typing could be a bit too much, wdyt?